packages/react-composites/src/composites/CallComposite/components/MediaGallery.tsx

```diff
@@ -60,11 +65,57 @@ export const MediaGallery = (props: MediaGalleryProps): JSX.Element => {
     };
   }, [cameraSwitcherCallback, cameraSwitcherCameras]);
 
+  const [augmentedParticipants, setAugmentedParticipants] = useState<VideoGalleryRemoteParticipant[] | undefined>(
+    undefined
+  );
+
+  // Async function to retrieve custom data model information to replace display name in remote tiles with camera on.
+  const fetchAvatarPersonaDataAsync = memoizeFunction(
+    async (
+      onFetchAvatarPersonaData: AvatarPersonaDataCallback,
+      participants: VideoGalleryRemoteParticipant[]
+    ): Promise<VideoGalleryRemoteParticipant[] | undefined> => {
+      if (props.onFetchAvatarPersonaData) {
+        participants.forEach(async (participant: VideoGalleryRemoteParticipant) => {
+          const newParticipantData = await onFetchAvatarPersonaData(participant.userId);
```

Comment:
- async/await doesn't work inside `forEach` loop. In this case it's just firing off multiple async calls but not really awaiting their results.
- You can use `for (const participant of participants)` instead.
- Better yet, use `Promise.all` which is much better than awaiting each async invocation
```tsx
await Promise.all(participants.map(async (participant) => {
  const newParticipantData = await onFetchAvatarPersonaData(participant.userId);
  participant.displayName = newParticipantData.text ? newParticipantData.text : participant.displayName;
}));
```

Reply:
And +1 to what @JamesBurnside  said. Mutating state variables inside components goes against our unidirectional data flow.
Better to introduce a component local state variable for `augmentedParticipants`